compendium/ConfigurationAdmin/test/TestAsyncWorkService.cpp

```diff
@@ -378,10 +354,9 @@ namespace test
 
     INSTANTIATE_TEST_SUITE_P(AsyncWorkServiceEndToEndParameterized,
                              TestAsyncWorkServiceEndToEnd,
```

Comment:
Inline execution is not guarenteed in our code base. In this case (in ConfigAdminImpl::performAsync), the posted task is done while holding the futuresMutex which is locked in the cleanupFutures call which always results in a deadlocked if inlined. Additionally, the AsyncWorkServiceStdAsync is not actually async if the returned future is not moved from, so this is essentially just another inline run

https://stackoverflow.com/questions/44654548/stdasync-doesnt-work-asynchronously#:~:text=If%20the%20std%3A%3Afuture%20obtained%20from%20std%3A%3Aasync%20is%20not%20moved%20from%20or%20bound%20to%20a%20reference%2C%20the%20destructor%20of%20the%20std%3A%3Afuture%20will%20block%20at%20the%20end%20of%20the%20full%20expression%20until%20the%20asynchronous%20operation%20completes%2C%20essentially%20making%20code%20...%20synchronous